tests/system/test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py

```diff
@@ -0,0 +1,197 @@
+'''
+
+import os
+import pytest
+import time
+from wazuh_testing import T_1, T_10
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.tools.file import read_yaml
+from wazuh_testing.tools.monitoring import HostMonitor
+from wazuh_testing.tools.system import HostManager
+from wazuh_testing.tools.file import replace_regex_in_file
+from system import (assign_agent_to_new_group, clean_cluster_logs, create_new_agent_group, delete_agent_group,
+                    restart_cluster)
+
+# pytestmark = [pytest.mark.cluster, pytest.mark.one_manager_agent_env]
+
+agent_conf_file = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                               '..', 'provisioning', 'one_manager_agent', 'roles', 'agent-role', 'files', 'ossec.conf')
+data_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+                              '..', 'provisioning', 'one_manager_agent', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+messages_path = os.path.join(local_path, 'data/messages.yaml')
+test_cases_yaml = read_yaml(os.path.join(data_path, 'cases_correct_merged_file_generation.yaml'))
+tmp_path = os.path.join(local_path, 'tmp')
+
+reset_files = {
+    'default': ['TestFile', 'TestFile2', 'EmptyFile', 'EmptyFile2', 'EmptyFile3', 'EmptyFile4', 'EmptyFile5',
+                'EmptyFile6', 'EmptyFile7', 'EmptyFile8', 'EmptyFile9', 'EmptyFile10'],
+    'TestGroup1': ['TestFileInTestGroup', 'TestFileInTestGroup2', 'EmptyFileInGroup', 'EmptyFileInGroup2',
+                   'EmptyFileInGroup3', 'EmptyFileInGroup4', 'EmptyFileInGroup5', 'EmptyFileInGroup6',
+                   'EmptyFileInGroup7', 'EmptyFileInGroup8', 'EmptyFileInGroup9', 'EmptyFileInGroup10']}
+testinfra_hosts = ['wazuh-manager', 'wazuh-agent1']
+
+
+@pytest.fixture()
+def environment_setting(test_case):
+
+    if test_case['metadata']['test_type'] == 'on_start':
+        host_manager.run_command(testinfra_hosts[0], f'{WAZUH_PATH}/bin/wazuh-control stop')
+    time.sleep(T_1)
+
+    yield
+
+    for file in reset_files['default']:
+        host_manager.run_command(testinfra_hosts[0], f'rm {WAZUH_PATH}/etc/shared/default/{file}.txt -f')
+    for file in reset_files['TestGroup1']:
+        host_manager.run_command(testinfra_hosts[0], f'rm {WAZUH_PATH}/etc/shared/TestGroup1/{file}.txt -f')
+    delete_agent_group(testinfra_hosts[0], 'TestGroup1', host_manager, 'api')
+    host_manager.run_command(testinfra_hosts[0], f'rm -r {WAZUH_PATH}/etc/shared/TestGroup1 -f')
+    create_new_agent_group(testinfra_hosts[0], 'TestGroup1', host_manager)
+    assign_agent_to_new_group(testinfra_hosts[0], 'TestGroup1',
+                              host_manager.run_command('wazuh-manager',
+                                                       f'cut -c 1-3 {WAZUH_PATH}/etc/client.keys'), host_manager)
```

Comment:
After the yield, the environment is usually cleaned, no groups should be created or assigned, as this would leave the environment dirty for the next test. A test should end with a clean environment.

Reply:
Changes done in https://github.com/wazuh/wazuh-qa/pull/4129/commits/5134d4c56b8ee81f07f1d17aec935dbd80a0d6d2